internal/agent/device/device_test.go

```diff
@@ -74,47 +76,72 @@ func TestSync(t *testing.T) {
 				mockHookManager *hook.MockManager,
 				mockAppManager *applications.MockManager,
 				mockLifecycleManager *lifecycle.MockManager,
+				mockPolicyManager *policy.MockManager,
+				mockSpecManager *spec.MockManager,
 			) {
 				nonRetryableHookError := errors.New("hook error")
 				gomock.InOrder(
+					mockSpecManager.EXPECT().GetDesired(ctx).Return(desired, false, nil),
+					mockSpecManager.EXPECT().Read(spec.Current).Return(current, nil),
+					mockPolicyManager.EXPECT().Sync(ctx, desired.Spec).Return(nil),
+					mockSpecManager.EXPECT().CheckPolicy(ctx, policy.Download, desired.Version()).Return(nil),
+					mockSpecManager.EXPECT().IsUpgrading().Return(true),
```

Comment:
note: we were using an actual spec manager before which didn't provide the same level of observability to the flow.